finetune/base.py

```diff
@@ -708,6 +709,41 @@ def finetune_grid_search_cv(cls, Xs, Y, *, n_splits, test_size, eval_fn=None, pr
 
         return max(aggregated_results, key=lambda x: x[1])[0]
 
+    def process_long_sequence(self, X, task, probas=False):
+        tasks = ['sequence_labeling', 'classification', 'regression']
+        assert task in tasks, 'invalid task for processing long sequences'
+        chunk_size = self.config.max_length - 2
+        step_size = chunk_size // 3
+        if task == 'sequence_labeling':
+            arr_encoded = list(itertools.chain.from_iterable(self.input_pipeline._text_to_ids([x]) for x in X))
+        else:
+            arr_encoded = list(itertools.chain.from_iterable(self.input_pipeline._text_to_ids(x) for x in X))
+        
+        labels, batch_probas = [], []
+        pred_keys = [PredictMode.NORMAL]
+        if task != 'regression':
+            pred_keys.append(PredictMode.PROBAS)
+        for pred in self._inference(X, predict_keys=[PredictMode.PROBAS, PredictMode.NORMAL], n_examples=len(arr_encoded)):
+            try:
+                labels.append(self.input_pipeline.label_encoder.inverse_transform([pred[PredictMode.NORMAL]]))
+            except ValueError:
```

Comment:
This try/except makes me nervous.  When is wrapping in the additional list necessary?  When is it not?